data/frameSimulator/include/ExcaliburFrame.h

```diff
@@ -0,0 +1,50 @@
+#ifndef FRAMESIMULATOR_EXCALIBURFRAME_H
+#define FRAMESIMULATOR_EXCALIBURFRAME_H
+
+#include <vector>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "Packet.h"
+
+namespace FrameSimulator {
+
+    class ExcaliburFrameSimulatorPlugin;
+
+    typedef std::vector<Packet> PacketList;
+
+    /** ExcaliburFrame class
+     *
+     * An excalibur frame and its packets; defines the SOF and EOF markers
+     * for reading an excalibur frame from a pcap file
+     */
+
+    class ExcaliburFrame {
+
+        friend class ExcaliburFrameSimulatorPlugin;
+
+    public:
+
+        ExcaliburFrame(const int& frameNum);
+
+        static const unsigned int SOF_marker = 1 << 31;
+        static const unsigned int EOF_marker = 1 << 30;
+        static const int num_subframes = 2;
+
```

Comment:
Some of this information I think is already defined in the Excalibur header file (and I think it is in common).  Is there any way we can use that header as an include to avoid having the same constants defined in two places?

Reply:
I had hard coded num_subframes to 2 (from copying the python script). I've now replaced with Excalibur::num_subframes from the Excalibur header file. This is indexed by bit depth. I've set an assumed bit depth of Excalibur::bitDepth1. Can the correct bit depth be deduced from the pcap files? Should it be an argument to the simulator?

Reply:
Good point, I'm not sure that it can be deduced although @timcnicholls will be able to confirm that.  I think that this may need to be an argument assuming that is OK.

Reply:
You can't deduce the bit depth from the PCAP file unfortunately. However, num_subframes is only != 2 when the bit depth is 24, in which case the system effectively produces two frames of two subframes each, since it is reading both 12 bit counters out in one go. I think it is perfectly OK for the simulator to just assume num_subframes=2 and operate like that all the time (like the python simulator does).

The knowledge that bitDepth24 has 4 subframes is only needed downstream in the FR/FP, when the decoder plugin explicitly merges 4 subframes into a single shared memory buffer to simplify processing.

Reply:
Great that simplifies it.